src/custom/components/SearchModal/CurrencySearch/useTokenSearch.ts

```diff
@@ -49,6 +49,18 @@ const tokensData = loadable(
   })
 )
 
+export function useTokenLogo(currency?: Currency | null): string | void {
+  const tokens = useAtomValue(useMemo(() => tokensData, [currency]))
+
+  if (!!currency && currency.isToken && tokens.state === 'hasData' && tokens.data) {
+    const token = tokens.data.data.find(
+      (token) => token.chainId === currency.chainId && token.address.toLowerCase() === currency.address.toLowerCase()
```

Comment:
why not saving `currency.address.toLowerCase()` in a const so u don't need to do that on every iteration of the loop